plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3314,6 +3310,32 @@ public Optional<ConnectorTableLayout> getNewTableLayout(ConnectorSession session
                         .build()));
     }
 
+    @Override
+    public Optional<ConnectorTableLayout> getLayoutForTableExecute(ConnectorSession session, ConnectorTableExecuteHandle executeHandle)
```

Comment:
@losipiuk @findepi It feels like there should be no downside in providing layout information to the engine (similar as for write, it is a responsibility of the engine to decide whether to use it or not). Do you see any potential issues with that?

Reply:
Not really.

Reply:
One problem I have with this code that we are replicating checks which we have already in `getTableHandleForOptimize`. Also code structure elsewhere assumes we can heve more that one table procedure. While here you assume the only one present is OPTIMIZE and make checks which are relevant for that one.

Reply:
I think once more procedures are added I would assume the tests would follow that are going to fail here. Not sure if it's worth adding explicit checks.